src/SignalR/common/Shared/JsonUtils.cs

```diff
@@ -114,6 +114,19 @@ public static void EnsureArrayStart(JsonTextReader reader)
             }
         }
 
+        public static bool ReadAsBoolean(JsonTextReader reader, string propertyName)
+        {
+            reader.Read();
+
+            if (reader.TokenType != JsonToken.Boolean || reader.Value == null)
+            {
+                throw new InvalidDataException($"Expected '{propertyName}' to be of type {JTokenType.Boolean}.");
+            }
+
+            // REVIEW: I'm trying to keep this similar to ReadAsInt32, but why not just call reader.ReadAsBoolean() and verify it's not null?
```

Comment:
I believe that Newtonsoft.Json is too nice and will read both `"true"` and `true` as boolean, but we'd prefer to be stricter and only allow `true`